src/main/java/org/jitsi/jigasi/transcription/AbstractTranscriptPublisher.java

```diff
@@ -450,26 +630,157 @@ BaseFormatter endedOn(TranscriptEvent event)
+            {
+                // we need to make sure the directory exists already
+                // so the recorder can write to it
+                createDirectoryIfNotExist(Paths.get(getLogDirPath(), dirName));
+
+                String format = getRecordingAudioFormat();
+                this.audioRecordingFileName =
+                   generateHardToGuessTimeString("",
+                        String.format(".%s", format));
+
+                String audioFilePath = Paths.get(getLogDirPath(), dirName,
+                    audioRecordingFileName).toString();
+
+                this.recorder = new RecorderImpl(device);
+                try
+                {
+                    this.recorder.start(format, audioFilePath);
+                }
+                catch (MediaException | IOException e)
+                {
+                    logger.error("Could not start recording", e);
+                }
+            }
+            else
+            {
+                logger.warn("receiving AudioMixerMediaDevice while not " +
+                    "requesting to record audio");
+            }
+        }
+
         @Override
-        public String getDescription()
+        public final synchronized void publish(Transcript transcript)
         {
-            return "Transcript will be available after the conference at " +
-                getBaseURL() + getFileName() + ".\n";
+            if (published)
+            {
+                return;
+            }
+            else
+            {
+                published = true;
+            }
+
+            if (this.recorder != null)
+            {
+                this.recorder.stop();
+            }
+
+            innerPublish(transcript);
+
+            maybeExecuteBashScripts();
         }
 
         /**
-         * Get the file name which will be used to store the {@link Transcript}
+         * Abstract method which needs to be implemented by children to publish
+         * the transcript and/or recording
          *
-         * @return the file name as a string
+         * @param transcript the Transcript file which needs to be published
+         */
+        protected abstract void innerPublish(Transcript transcript);
+
+        /**
+         * Get the filename which was used to store the audio recording to
+         *
+         * @return the filename
          */
-        protected abstract String getFileName();
+        public String getAudioRecordingFileName()
+        {
+            return audioRecordingFileName;
+        }
+
+        /**
+         * Execute all given scripts by
+         * {@link this#getPathsToScriptsToExecute()} ()} when
+         * {@link this#executeScripts()} returns true
+         */
+        private void maybeExecuteBashScripts()
+        {
+            if (executeScripts())
+            {
+                Path dirPath = Paths.get(getLogDirPath(), dirName)
+                    .toAbsolutePath();
+
+                for(String path : getPathsToScriptsToExecute())
```

Comment:
The naming of `path` and `scriptPath` is slightly confusing -- they are both paths to the script. I'd suggest `path` and `absPath`, or `scriptPath` and `absScriptPath`